packages/wonder-blocks-grid/components/row.js

```diff
@@ -31,108 +53,89 @@ import Gutter from "./gutter.js";
-          }) => React.Node),
-    /** The styling to apply to the row. */
-    style?: StyleType,
-    /**
-     * The size of the media layout being used. Populated by MediaLayoutWrapper.
-     * @ignore
-     */
-    mediaSize: MediaSize,
-    /**
-     * The current media layout spec being used. Populated by MediaLayoutWrapper.
-     * @ignore
-     */
-    mediaSpec: MediaSpec,
-}> {
+export default class Row extends React.Component<Props> {
     static defaultProps = {
-        small: false,
-        medium: false,
-        large: false,
+        mediaQuery: "all",
     };
 
     render() {
-        const {style, mediaSize, mediaSpec, children} = this.props;
-        const {marginWidth, maxWidth, totalColumns} = mediaSpec[mediaSize];
-        const shouldDisplay = matchesSize(this.props, mediaSize);
+        const {style, children} = this.props;
+
+        return (
+            <MediaLayout>
+                {({mediaSize, mediaSpec}) => {
+                    const {marginWidth, maxWidth, totalColumns} = mediaSpec[
+                        mediaSize
+                    ];
+                    const shouldDisplay = queryMatchesSize(
+                        this.props.mediaQuery,
+                        mediaSize,
+                    );
 
-        // Don't render the row if it's been disabled at this size
-        if (!shouldDisplay) {
-            return null;
-        }
+                    // Don't render the row if it's been disabled at this size
+                    if (!shouldDisplay) {
+                        return null;
+                    }
 
-        let contents = children;
+                    let contents = children;
 
-        // If the contents are a function then we call it with the mediaSize and
-        // totalColumns properties and render the return value.
-        if (typeof contents === "function") {
-            contents = contents({mediaSize, totalColumns});
-        }
+                    // If the contents are a function then we call it with the mediaSize and
+                    // totalColumns properties and render the return value.
+                    if (typeof contents === "function") {
+                        contents = contents({mediaSize, totalColumns});
+                    }
 
-        contents = React.Children.toArray(contents);
+                    contents = React.Children.toArray(contents);
 
-        // Go through all of the contents and pre-emptively remove anything
-        // that shouldn't be rendered and insert Gutters inbetween everything
-        // that is still visible.
-        const filteredContents = [];
-        let hasVisibleCell = false;
+                    // Go through all of the contents and pre-emptively remove anything
+                    // that shouldn't be rendered and insert Gutters inbetween everything
+                    // that is still visible.
+                    const filteredContents = [];
+                    let hasVisibleCell = false;
```

Comment:
Seems like this should be its own method for sure.

Reply:
In fact, isn't this just a placeholder for `filteredContents.length > 0`?

Reply:
yeah, I think this could be reworked to filter first and then intersperse `<Gutter>` elements in between the `<Cells>` afterwards.